test/test_options.py

```diff
@@ -75,3 +76,12 @@ def test_sorting_factor_does_not_exist1(self, _, mock_exit):
     def test_will_include_ext_args(self):
         options = parse_args(['lizard', '--ND', '2', '-End'])
         self.assertEqual(2, options.ND)
+
+    @patch('lizard.auto_read', create=True)
+    @patch.object(lizard, 'print_result')
+    @patch.object(sys, 'open')
+    def test_max_recursion_depth(self, _, print_result, mock_open):
+        mock_open.return_value = "void main(){return;}"
+        print_result.side_effect = lizard.silent_printer
+        lizard.main(['lizard', '--max-recursion-depth', '10000'])
+        self.assertEqual(10_000, sys.getrecursionlimit())
```

Comment:
How about mocking `sys.setrecursionlimit`?

```suggestion
    @patch.object(sys, 'setrecursionlimit')
    def test_max_recursion_depth(self, setrecursionlimit, print_result, mock_open):
        mock_open.return_value = "void main(){return;}"
        print_result.side_effect = lizard.silent_printer
        lizard.main(['lizard', '--max-recursion-depth', '10000'])
        setrecursionlimit.assert_called_with(10_000)
```

Reply:
Thanks for your good suggestion. fix: 7eb64b3b0d7a5ba9581c7f82dcbeee01ced08254